Crashlytics/Crashlytics/Models/Record/FIRCLSRecordAdapter.m

```diff
@@ -0,0 +1,147 @@
+#import "FIRCLSRecordKeyValue.h"
+#import "FIRCLSRecordProcessStats.h"
+#import "FIRCLSRecordRegister.h"
+#import "FIRCLSRecordRuntime.h"
+#import "FIRCLSRecordSignal.h"
+#import "FIRCLSRecordStorage.h"
+#import "FIRCLSRecordThread.h"
+
+@interface FIRCLSRecordAdapter ()
+
+@property(nonatomic, strong) NSString *folderPath;
+
+@property(nonatomic, strong) FIRCLSRecordSignal *signal;
+@property(nonatomic, strong) NSArray<FIRCLSRecordThread *> *threads;
+@property(nonatomic, strong) FIRCLSRecordProcessStats *processStats;
+@property(nonatomic, strong) FIRCLSRecordStorage *storage;
+@property(nonatomic, strong) NSArray<FIRCLSRecordBinaryImage *> *binaryImages;
+@property(nonatomic, strong) FIRCLSRecordRuntime *runtime;
+@property(nonatomic, strong) FIRCLSRecordIdentity *identity;
+@property(nonatomic, strong) FIRCLSRecordHost *host;
+@property(nonatomic, strong) FIRCLSRecordApplication *application;
+@property(nonatomic, strong) FIRCLSRecordExecutable *executable;
+@property(nonatomic, strong) NSArray<FIRCLSRecordKeyValue *> *keyValues;
+
+@end
+
+@implementation FIRCLSRecordAdapter
+
+- (instancetype)initWithPath:(NSString *)folderPath {
+  self = [super init];
+  if (self) {
+    _folderPath = folderPath;
+
+    [self loadBinaryImagesFile];
+    [self loadMetaDataFile];
+    [self loadSignalFile];
+    [self loadKeyValuesFile];
+  }
+  return self;
+}
+
+- (void)loadBinaryImagesFile {
+  NSString *path = [self.folderPath stringByAppendingPathComponent:CLSReportBinaryImageFile];
+  self.binaryImages = [FIRCLSRecordBinaryImage
+      binaryImagesFromDictionaries:[FIRCLSRecordAdapter dictionariesFromEachLineOfFile:path]];
+}
+
+- (void)loadMetaDataFile {
+  NSString *path = [self.folderPath stringByAppendingPathComponent:CLSReportMetadataFile];
+  NSDictionary *dict = [FIRCLSRecordAdapter combinedDictionariesFromFilePath:path];
+
+  self.identity = [[FIRCLSRecordIdentity alloc] initWithDict:dict[@"identity"]];
+  self.host = [[FIRCLSRecordHost alloc] initWithDict:dict[@"host"]];
+  self.application = [[FIRCLSRecordApplication alloc] initWithDict:dict[@"application"]];
+  self.executable = [[FIRCLSRecordExecutable alloc] initWithDict:dict[@"executable"]];
+}
+
+- (void)loadSignalFile {
+  NSString *path = [self.folderPath stringByAppendingPathComponent:CLSReportSignalFile];
+  NSDictionary *dicts = [FIRCLSRecordAdapter combinedDictionariesFromFilePath:path];
+
+  self.signal = [[FIRCLSRecordSignal alloc] initWithDict:dicts[@"signal"]];
+  self.runtime = [[FIRCLSRecordRuntime alloc] initWithDict:dicts[@"runtime"]];
+  self.processStats = [[FIRCLSRecordProcessStats alloc] initWithDict:dicts[@"process_stats"]];
+  self.storage = [[FIRCLSRecordStorage alloc] initWithDict:dicts[@"storage"]];
+  self.threads = [FIRCLSRecordThread threadsFromDictionaries:dicts[@"threads"]
+                                                   withNames:dicts[@"thread_names"]
+                                                 withRuntime:self.runtime];
```

Comment:
Can you add a quick comment on why runtime is passed in here?

Reply:
Sure, I'll add a comment. The main reason is that we set the thread's objc_selector_name with the runtime's info.